### PR TITLE
[JENKINS-72845]: Fix status icon animation display on Safari

### PR DESCRIPTION
**Pull request by Lmh-java**
See [JENKINS-72845](https://issues.jenkins.io/browse/JENKINS-72845).

The status icon animations are not properly shown on Safari if the user zooms in/out (`Command + "+" or "-"`). Some bugs related to the Safari browser were reported a few years ago, and it seems this is still not fixed. The main issue is that Safari cannot properly calculate the `center` value of the property `transform-origin`, it behaves differently between Safari and Chrome.

~By refactoring the structure of SVG icons and using SVG animations instead of CSS animations, this display problem is solved.~
I re-made these icon SVGs and centerized every SVG component to an enlarged view window. Therefore, we can directly set `transform-origin: 0 0;` to make sure the origin pivot is not changing in any cases, so Safari can properly locate the animation.

PS: This is my first time contributing to Jenkins. If I am doing something wrong,  please bear with me. Any advice is truly appreciated. :)

<details>
<summary>Screenshot before:</summary>

Before

This is just an example of one status icon. All other animations are similar.
It also happens in the build history section and the timeline section.
</details>

<details>
<summary>Screenshot after:</summary>

After

All occurrences of this are fixed.
</details>

### Testing done
1. Manual testing on Safari 16.5.1, Chrome 123.0.6312.106.
2. Lightmode unit tests

### Proposed changelog entries

- Fix status icon animation display on Safari.

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Comment by daniel-beck** [comment]
Thanks!

Does this mean https://github.com/jenkinsci/jenkins/blob/bb45b8722bd3d8c3122cfd481778084d1b8db612/war/src/main/scss/base/_style.scss#L871-L900 could be deleted? That also references reduced motion, is this something that is being removed with this change?

**Comment by Lmh-java** [comment]
> Thanks!
> 
> Does this mean
> 
> https://github.com/jenkinsci/jenkins/blob/bb45b8722bd3d8c3122cfd481778084d1b8db612/war/src/main/scss/base/_style.scss#L871-L900
> 
> could be deleted? That also references reduced motion, is this something that is being removed with this change?

Hi! I didn't notice `prefers-reduced-motion` at first glance, I will aim to remove this with this PR as well :).

**Comment by Lmh-java** [comment]
> Thanks!
> 
> Does this mean
> 
> https://github.com/jenkinsci/jenkins/blob/bb45b8722bd3d8c3122cfd481778084d1b8db612/war/src/main/scss/base/_style.scss#L871-L900
> 
> could be deleted? That also references reduced motion, is this something that is being removed with this change?

Solved by a new method, please take a look :). In the new solution, I am still using the css properties to animate, so no need to remove the original css.

**Comment by Lmh-java** [comment]
@timja Hi, sorry for pinging. Could you please take a look at this PR please. Thanks.

**Comment by Lmh-java** [comment]
@timja @daniel-beck Please take a look when you have time.

**Review by daniel-beck — APPROVED** [review]
Looking at the "Icon legend" view/popup, this seems to work well in Safari, Chrome and Firefox now. I don't know enough about SVG to judge the technical quality of the fix or predict other problems this might cause.

**Review by timja — APPROVED** [review]
https://github.com/jenkinsci/jenkins/pull/9123#pullrequestreview-2013184649
Same, but this fix has been sitting here long enough and if it works lets go for it.

----

/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Comment by Lmh-java** [comment]
Thank you for both of your responses.

Here is a little more explanation of this change.

This animation is a rotating around one pivot. Specifically, a part of the svg image is rotating around that pivot. I noticed that if we zoom in/out in safari, the new pivot position will not be re-calculated. Therefore, it will still rotate around the old position. This seems to be a bug with Safari since many years ago. To fix this problem, I mapped all the coordinates of the original svg to a new coordinate system, which centered at (0, 0). Therefore, if we zoom in/out, the position of the pivot will not be changed. Without re-calculating the new coordinate, this animation will not shift.

**Pull request merged**
